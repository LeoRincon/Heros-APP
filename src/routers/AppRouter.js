import React from 'react';
import { BrowserRouter, Routes, Route } from 'react-router-dom';

import { LoginScreen } from '../components/login';
import { DashboardRouter } from './';

export const AppRouter = () => {
  return (
    <BrowserRouter>
      <div>
        <Routes>
          <Route path='/*' element={<DashboardRouter />} />
          <Route exact path='/login' element={<LoginScreen />} />
        </Routes>
      </div>
    </BrowserRouter>
  );
};
