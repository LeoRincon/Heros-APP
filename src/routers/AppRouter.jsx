import React from 'react';
import { BrowserRouter, Routes, Route } from 'react-router-dom';

import { LoginScreen } from '../components/login';
import { DashboardRouter } from '.';
import { PrivateRoute } from '.';
import { PublicRoute } from '.';

export const AppRouter = () => {
  return (
    <BrowserRouter>
      <Routes>
        <Route
          path='/login'
          element={
            <PublicRoute>
              <LoginScreen />
            </PublicRoute>
          }
        />
        <Route
          path='/*'
          element={
            <PrivateRoute>
              <DashboardRouter />
            </PrivateRoute>
          }
        />
      </Routes>
    </BrowserRouter>
  );
};
