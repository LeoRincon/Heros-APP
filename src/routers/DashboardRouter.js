import React from 'react';
import { Navbar } from '../components/ui/NavBar';
import { Routes, Route } from 'react-router-dom';

import { MarvelScreen } from '../pages/marvel';
import { DcScreen } from '../pages/dcScreen';
import { HeroScreen } from '../components/heroes';

export const DashboardRouter = () => {
  return (
    <>
      <Navbar />
      <div className='container mt-3'>
        <Routes>
          <Route exact path='/marvel' element={<MarvelScreen />} />
          <Route exact path='/heroe/:heroeId' element={<HeroScreen />} />
          <Route exact path='/dc' element={<DcScreen />} />
          <Route path='*' element={<MarvelScreen />} />
        </Routes>
      </div>
    </>
  );
};
