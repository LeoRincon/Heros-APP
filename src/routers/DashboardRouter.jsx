import React from 'react';
import { Navbar } from '../components/ui/NavBar';
import { Routes, Route } from 'react-router-dom';

import { MarvelScreen } from '../pages/marvel';
import { DcScreen } from '../pages/dcScreen';
import { HeroScreen } from '../components/heroes';
import { SearchScreen } from '../pages/search/SearchScreen';

export const DashboardRouter = () => {
  return (
    <>
      <Navbar />
      <div className='container mt-3'>
        <Routes>
          <Route exact path='/marvel' element={<MarvelScreen />} />
          <Route exact path='/hero/:heroeId' element={<HeroScreen />} />
          <Route exact path='/dc' element={<DcScreen />} />
          <Route exact path='/search' element={<SearchScreen />} />
          <Route path='*' element={<MarvelScreen />} />
        </Routes>
      </div>
    </>
  );
};
