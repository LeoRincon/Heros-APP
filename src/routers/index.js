export { DashboardRouter } from './DashboardRouter';
export { PrivateRoute } from './PrivateRoute';
export { AppRouter } from './AppRouter';
export { PublicRoute } from './PublicRoute';
