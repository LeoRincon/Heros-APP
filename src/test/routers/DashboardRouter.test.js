import { mount } from 'enzyme';
import { DashboardRouter } from '../../routers/DashboardRouter';
import { AuthContext } from '../../auth/AuthContext';
import { MemoryRouter } from 'react-router-dom';

describe('tests in <DashboardRouter /> when is called', () => {
  const contextValue = {
    user: {
      name: 'Pepe',
      logged: true,
    },
  };
  test('should be displayed correctly Marvel Screen', () => {
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <MemoryRouter initialEntries={['/']}>
          <DashboardRouter />
        </MemoryRouter>
      </AuthContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.text-info').text().trim()).toBe('Pepe');
    expect(wrapper.find('h1').text().trim()).toBe('Marvel Screen');
  });

  test('should be displayed correctly DC Screen', () => {
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <MemoryRouter initialEntries={['/dc']}>
          <DashboardRouter />
        </MemoryRouter>
      </AuthContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('h1').text().trim()).toBe('Dc Screen');
  });
});
