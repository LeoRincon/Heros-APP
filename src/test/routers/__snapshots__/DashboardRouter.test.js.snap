// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in <DashboardRouter /> when is called should be displayed correctly DC Screen 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          className="nav-item nav-link "
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          className="nav-item nav-link "
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end px-4"
    >
      <nav
        className="navbar-nav ml-auto"
      >
        <span
          className="nav-item nav-link text-info"
        >
          Pepe
        </span>
        <button
          className="nav-item nav-link btn"
          exact="true"
          onClick={[Function]}
          to="/login"
        >
          Logout
        </button>
      </nav>
    </div>
  </nav>,
  <div
    className="container mt-3"
  >
    <div>
      <h1>
        Dc Screen
      </h1>
      <div
        className="row animate__animated animate__fadeIn"
      >
        <div
          className="card ms-3 animate__animated animate__fadeIn"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Batman"
                className="card-img"
                src="./assets/heroes/dc-batman.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Batman
                   (
                  Bruce Wayne
                  )
                </h5>
                <p
                  className="card-text mb-5"
                >
                   
                  First Appearance:
                  <small
                    className="text-muted"
                  >
                     
                    Detective Comics #27
                  </small>
                </p>
                <a
                  className="btn btn-primary "
                  href="/hero/dc-batman"
                  onClick={[Function]}
                >
                  Ver más
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3 animate__animated animate__fadeIn"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Superman"
                className="card-img"
                src="./assets/heroes/dc-superman.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Superman
                   (
                  Kal-El
                  )
                </h5>
                <p
                  className="card-text mb-5"
                >
                   
                  First Appearance:
                  <small
                    className="text-muted"
                  >
                     
                    Action Comics #1
                  </small>
                </p>
                <a
                  className="btn btn-primary "
                  href="/hero/dc-superman"
                  onClick={[Function]}
                >
                  Ver más
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3 animate__animated animate__fadeIn"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Flash"
                className="card-img"
                src="./assets/heroes/dc-flash.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Flash
                   (
                  Jay Garrick
                  )
                </h5>
                <p
                  className="card-text"
                >
                  Jay Garrick, Barry Allen, Wally West, Bart Allen
                </p>
                <p
                  className="card-text mb-5"
                >
                   
                  First Appearance:
                  <small
                    className="text-muted"
                  >
                     
                    Flash Comics #1
                  </small>
                </p>
                <a
                  className="btn btn-primary "
                  href="/hero/dc-flash"
                  onClick={[Function]}
                >
                  Ver más
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3 animate__animated animate__fadeIn"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Green Lantern"
                className="card-img"
                src="./assets/heroes/dc-green.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Green Lantern
                   (
                  Alan Scott
                  )
                </h5>
                <p
                  className="card-text"
                >
                  Alan Scott, Hal Jordan, Guy Gardner, John Stewart, Kyle Raynor, Jade, Sinestro, Simon Baz
                </p>
                <p
                  className="card-text mb-5"
                >
                   
                  First Appearance:
                  <small
                    className="text-muted"
                  >
                     
                    All-American Comics #16
                  </small>
                </p>
                <a
                  className="btn btn-primary "
                  href="/hero/dc-green"
                  onClick={[Function]}
                >
                  Ver más
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3 animate__animated animate__fadeIn"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Green Arrow"
                className="card-img"
                src="./assets/heroes/dc-arrow.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Green Arrow
                   (
                  Oliver Queen
                  )
                </h5>
                <p
                  className="card-text mb-5"
                >
                   
                  First Appearance:
                  <small
                    className="text-muted"
                  >
                     
                    More Fun Comics #73
                  </small>
                </p>
                <a
                  className="btn btn-primary "
                  href="/hero/dc-arrow"
                  onClick={[Function]}
                >
                  Ver más
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3 animate__animated animate__fadeIn"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Wonder Woman"
                className="card-img"
                src="./assets/heroes/dc-wonder.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Wonder Woman
                   (
                  Princess Diana
                  )
                </h5>
                <p
                  className="card-text mb-5"
                >
                   
                  First Appearance:
                  <small
                    className="text-muted"
                  >
                     
                    All Star Comics #8
                  </small>
                </p>
                <a
                  className="btn btn-primary "
                  href="/hero/dc-wonder"
                  onClick={[Function]}
                >
                  Ver más
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3 animate__animated animate__fadeIn"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Martian Manhunter"
                className="card-img"
                src="./assets/heroes/dc-martian.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Martian Manhunter
                   (
                  J'onn J'onzz
                  )
                </h5>
                <p
                  className="card-text"
                >
                  Martian Manhunter
                </p>
                <p
                  className="card-text mb-5"
                >
                   
                  First Appearance:
                  <small
                    className="text-muted"
                  >
                     
                    Detective Comics #225
                  </small>
                </p>
                <a
                  className="btn btn-primary "
                  href="/hero/dc-martian"
                  onClick={[Function]}
                >
                  Ver más
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3 animate__animated animate__fadeIn"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Robin/Nightwing"
                className="card-img"
                src="./assets/heroes/dc-robin.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Robin/Nightwing
                   (
                  Dick Grayson
                  )
                </h5>
                <p
                  className="card-text mb-5"
                >
                   
                  First Appearance:
                  <small
                    className="text-muted"
                  >
                     
                    Detective Comics #38
                  </small>
                </p>
                <a
                  className="btn btn-primary "
                  href="/hero/dc-robin"
                  onClick={[Function]}
                >
                  Ver más
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3 animate__animated animate__fadeIn"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Blue Beetle"
                className="card-img"
                src="./assets/heroes/dc-blue.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Blue Beetle
                   (
                  Dan Garret
                  )
                </h5>
                <p
                  className="card-text"
                >
                  Dan Garret, Ted Kord, Jaime Reyes
                </p>
                <p
                  className="card-text mb-5"
                >
                   
                  First Appearance:
                  <small
                    className="text-muted"
                  >
                     
                    Mystery Men Comics #1
                  </small>
                </p>
                <a
                  className="btn btn-primary "
                  href="/hero/dc-blue"
                  onClick={[Function]}
                >
                  Ver más
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3 animate__animated animate__fadeIn"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Black Canary"
                className="card-img"
                src="./assets/heroes/dc-black.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Black Canary
                   (
                  Dinah Drake
                  )
                </h5>
                <p
                  className="card-text"
                >
                  Dinah Drake, Dinah Lance
                </p>
                <p
                  className="card-text mb-5"
                >
                   
                  First Appearance:
                  <small
                    className="text-muted"
                  >
                     
                    Flash Comics #86
                  </small>
                </p>
                <a
                  className="btn btn-primary "
                  href="/hero/dc-black"
                  onClick={[Function]}
                >
                  Ver más
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`tests in <DashboardRouter /> when is called should be displayed correctly Marvel Screen 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          className="nav-item nav-link "
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          className="nav-item nav-link "
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          className="nav-item nav-link "
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end px-4"
    >
      <nav
        className="navbar-nav ml-auto"
      >
        <span
          className="nav-item nav-link text-info"
        >
          Pepe
        </span>
        <button
          className="nav-item nav-link btn"
          exact="true"
          onClick={[Function]}
          to="/login"
        >
          Logout
        </button>
      </nav>
    </div>
  </nav>,
  <div
    className="container mt-3"
  >
    <div>
      <h1>
        Marvel Screen
      </h1>
      <div
        className="row animate__animated animate__fadeIn"
      >
        <div
          className="card ms-3 animate__animated animate__fadeIn"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Spider Man"
                className="card-img"
                src="./assets/heroes/marvel-spider.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Spider Man
                   (
                  Peter Parker
                  )
                </h5>
                <p
                  className="card-text mb-5"
                >
                   
                  First Appearance:
                  <small
                    className="text-muted"
                  >
                     
                    Amazing Fantasy #15
                  </small>
                </p>
                <a
                  className="btn btn-primary "
                  href="/hero/marvel-spider"
                  onClick={[Function]}
                >
                  Ver más
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3 animate__animated animate__fadeIn"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Captain America"
                className="card-img"
                src="./assets/heroes/marvel-captain.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Captain America
                   (
                  Steve Rogers
                  )
                </h5>
                <p
                  className="card-text mb-5"
                >
                   
                  First Appearance:
                  <small
                    className="text-muted"
                  >
                     
                    Captain America Comics #1
                  </small>
                </p>
                <a
                  className="btn btn-primary "
                  href="/hero/marvel-captain"
                  onClick={[Function]}
                >
                  Ver más
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3 animate__animated animate__fadeIn"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Iron Man"
                className="card-img"
                src="./assets/heroes/marvel-iron.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Iron Man
                   (
                  Tony Stark
                  )
                </h5>
                <p
                  className="card-text mb-5"
                >
                   
                  First Appearance:
                  <small
                    className="text-muted"
                  >
                     
                    Tales of Suspense #39
                  </small>
                </p>
                <a
                  className="btn btn-primary "
                  href="/hero/marvel-iron"
                  onClick={[Function]}
                >
                  Ver más
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3 animate__animated animate__fadeIn"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Thor"
                className="card-img"
                src="./assets/heroes/marvel-thor.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Thor
                   (
                  Thor Odinson
                  )
                </h5>
                <p
                  className="card-text mb-5"
                >
                   
                  First Appearance:
                  <small
                    className="text-muted"
                  >
                     
                    Journey into Myster #83
                  </small>
                </p>
                <a
                  className="btn btn-primary "
                  href="/hero/marvel-thor"
                  onClick={[Function]}
                >
                  Ver más
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3 animate__animated animate__fadeIn"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Hulk"
                className="card-img"
                src="./assets/heroes/marvel-hulk.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Hulk
                   (
                  Bruce Banner
                  )
                </h5>
                <p
                  className="card-text mb-5"
                >
                   
                  First Appearance:
                  <small
                    className="text-muted"
                  >
                     
                    The Incredible Hulk #1
                  </small>
                </p>
                <a
                  className="btn btn-primary "
                  href="/hero/marvel-hulk"
                  onClick={[Function]}
                >
                  Ver más
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3 animate__animated animate__fadeIn"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Wolverine"
                className="card-img"
                src="./assets/heroes/marvel-wolverine.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Wolverine
                   (
                  James Howlett
                  )
                </h5>
                <p
                  className="card-text mb-5"
                >
                   
                  First Appearance:
                  <small
                    className="text-muted"
                  >
                     
                    The Incredible Hulk #180
                  </small>
                </p>
                <a
                  className="btn btn-primary "
                  href="/hero/marvel-wolverine"
                  onClick={[Function]}
                >
                  Ver más
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3 animate__animated animate__fadeIn"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Daredevil"
                className="card-img"
                src="./assets/heroes/marvel-daredevil.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Daredevil
                   (
                  Matthew Michael Murdock
                  )
                </h5>
                <p
                  className="card-text mb-5"
                >
                   
                  First Appearance:
                  <small
                    className="text-muted"
                  >
                     
                    Daredevil #1
                  </small>
                </p>
                <a
                  className="btn btn-primary "
                  href="/hero/marvel-daredevil"
                  onClick={[Function]}
                >
                  Ver más
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3 animate__animated animate__fadeIn"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Hawkeye"
                className="card-img"
                src="./assets/heroes/marvel-hawkeye.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Hawkeye
                   (
                  Clinton Francis Barton
                  )
                </h5>
                <p
                  className="card-text mb-5"
                >
                   
                  First Appearance:
                  <small
                    className="text-muted"
                  >
                     
                    Tales of Suspense #57
                  </small>
                </p>
                <a
                  className="btn btn-primary "
                  href="/hero/marvel-hawkeye"
                  onClick={[Function]}
                >
                  Ver más
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3 animate__animated animate__fadeIn"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Cyclops"
                className="card-img"
                src="./assets/heroes/marvel-cyclops.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Cyclops
                   (
                  Scott Summers
                  )
                </h5>
                <p
                  className="card-text mb-5"
                >
                   
                  First Appearance:
                  <small
                    className="text-muted"
                  >
                     
                    X-Men #1
                  </small>
                </p>
                <a
                  className="btn btn-primary "
                  href="/hero/marvel-cyclops"
                  onClick={[Function]}
                >
                  Ver más
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3 animate__animated animate__fadeIn"
          style={
            Object {
              "maxWidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Silver Surfer"
                className="card-img"
                src="./assets/heroes/marvel-silver.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Silver Surfer
                   (
                  Norrin Radd
                  )
                </h5>
                <p
                  className="card-text mb-5"
                >
                   
                  First Appearance:
                  <small
                    className="text-muted"
                  >
                     
                    The Fantastic Four #48
                  </small>
                </p>
                <a
                  className="btn btn-primary "
                  href="/hero/marvel-silver"
                  onClick={[Function]}
                >
                  Ver más
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
